### DISCUSSION TITLE
How work new_chat_members and left_chat_member

### DISCUSSION THREAD

**Discussion by ALLADMINdotRU** [eternnoir/pyTelegramBotAPI#1875] [Q&A] [answered]
I'm trying to exit and enter the chat, there is no response from the function.
Can you provide a working example?

@bot.message_handler(content_types=['left_chat_member'])           
def new_member(message):
    print('left chat')

**Comment by Badiboy** [comment]
It's fine, should work. Check that bot is admin, not in privacy mode etc.

Btw, left_chat_member appears not in all groups/channels (I do not remeber details).
Handle also chat_member_handler:
```
        @bot.chat_member_handler()
        def test_chat_member(chat_member_updated):
            ...
```
May be you will see events you are waiting for there.

**Reply by ALLADMINdotRU** [reply]
not work

try entering and exiting the chat five times.

**Comment by coder2020official** [comment]
You should pass a list of updates to infinity_polling including chat_member update type to use chat_member_handler.
Something like:
```
bot.infinity_polling(update_types=['chat_member', 'message'])
```

Add any other update types from [here](https://core.telegram.org/bots/api#update) if needed;
Or use util.update_types, which contains all update types.

**Reply by coder2020official** [reply]
And bot should be an admin in the chat.

**Reply by ALLADMINdotRU** [reply]
do you have a work code?

**Reply by coder2020official** [reply]
Just look at the thing badiboy have suggested and combine it with mine

**Reply by ALLADMINdotRU** [reply]
Understand, you don't have the work code

**Reply by coder2020official** [reply]
🤦  I've told you.
Is this so hard:

```
from telebot import TeleBot
from telebot.types import ChatMemberUpdated
from telebot.utils import update_types

@bot.chat_member_handler()
def test_chat_member(chat_member_updated: ChatMemberUpdated):
    if chat_member_updated.new_chat_member.status == 'left':
        # ur code

bot.infinity_polling(allowed_updates=update_types)
```

**Reply by ALLADMINdotRU** [reply]
unfortunately this is not a working code

**Reply by coder2020official** [reply]
unfortunately you are not able to use it correctly

**Reply by ALLADMINdotRU** [reply]
possible, but the fact is that when entering or leaving the chat, no events are processed

**Reply by majidhd001** [reply]
its wooooorkkkkk . kiss you

**Reply by majidhd001** [reply]
*from telebot.util import update_types

**Comment by ALLADMINdotRU** [comment]
again and it doesn't work


```
import telebot
from telebot.util import update_types
from telebot.types import ChatMemberUpdated

bot = telebot.TeleBot('522895') 

@bot.chat_member_handler()
def test_chat_member(chat_member_updated: ChatMemberUpdated):
    print(chat_member_updated.new_chat_member.status)


@bot.my_chat_member_handler()
def test_chat_member2(chat_member_updated: ChatMemberUpdated):
    print(chat_member_updated.new_chat_member.status)

bot.infinity_polling(none_stop=True, interval=0, allowed_updates=update_types) 

```

**Reply by coder2020official** [reply]
bot should be an admin in the chat to receive this type of update

**Reply by ALLADMINdotRU** [reply]
how to do it?

**Reply by ALLADMINdotRU** [reply]
> how to do it?
although I’ll figure it out myself, these are already telegram settings
but the strange fact is that having the key, the bot is not an admin

**Reply by ALLADMINdotRU** [reply]
so stop
the problem is that I need to know who was added to the bot as a friend (and not to the channel)

**Reply by ALLADMINdotRU** [reply]
and the bot is already an administrator in itself, what else do you need?

**Comment by ALLADMINdotRU** [comment]
although the end of the discussion I'm tired
try aiogram

**Reply by coder2020official** [reply]
lol aiogram won't help you in solving Ur question if u r dumb

**Reply by ALLADMINdotRU** [reply]
ho-ho)) look who's Talking

Unfortunately, this framework is not intuitive. 
It's easier for me to parse the json response and get the result than looking for working code here (which doesn't exist by the way)
I hope the iogram is devoid of this shortcoming.

**Reply by ALLADMINdotRU** [reply]
> lol aiogram won't help you in solving Ur question if u r dumb
show me the working code, all these sketches on paper and I can do it.

**Reply by coder2020official** [reply]
Good luck then, lol.

**Comment by coder2020official** [comment, marked as answer]
BTW, we have CUSTOM_REQUEST_SENDER in apihelper file to send request with given params manually, but as u r switching to aio, this answer is probably useless))

**Comment by MaGua-Bishop** [comment]
Your problem is easy to solve, I added group chat filter in 'message_handler' and it works. Here is the code I worked on
`@bot.message_handler(func=lambda message: True, chat_types=['group', 'supergroup'], content_types=['new_chat_members'])
def new_chat_members(message):
    print(message)`

By the way, how do you know which invitation link a user used to enter a group chat, or who sent the link? I'm sorry for my poor English
